src/tests/analyzers_conditional_test.py

```diff
@@ -0,0 +1,58 @@
+import ast
+import tempfile
+from functools import partial
+
+from tryceratops import analyzers
+from tryceratops.analyzers.conditional import STANDARD_NON_TYPE_ERROR_IDS
+from tryceratops.violations import codes
+
+from .analyzer_helpers import assert_violation, read_sample
+
+
+def test_prefer_type_error():
+    analyzer = analyzers.conditional.PreferTypeErrorAnalyzer()
+
+    assert_type_error = partial(
+        assert_violation, codes.PREFER_TYPE_ERROR[0], codes.PREFER_TYPE_ERROR[1]
+    )
+
+    for non_type_exception in STANDARD_NON_TYPE_ERROR_IDS:
+        with tempfile.NamedTemporaryFile(mode="r+") as tmp:
+            file_content = (
+                "if isinstance(some_arg, int):\n"
+                "   pass\n"
+                "else:\n"
+                f'   raise {non_type_exception}("...")'
+            )
+            tmp.write(file_content)
+            tmp.seek(0)
+            content = tmp.read()
+
+            tmp_tree = ast.parse(content)
+            tmp_violations = analyzer.check(tmp_tree, "filename")
+
+            assert len(tmp_violations) == 1
+            assert_type_error(4, 9, tmp_violations[0])
```

Comment:
@sbrugman is this what you had in mind when you mentioned [parametrizing these tests](https://github.com/guilatrova/tryceratops/pull/34#discussion_r740427039)?

Reply:
I think he meant to try [`pytest.mark.parametrize`](https://docs.pytest.org/en/6.2.x/parametrize.html). But I'm not confident it would make the code any easier.

Reply:
Ah I see, thanks